lib/reply.js

```diff
@@ -302,6 +307,80 @@ Reply.prototype.code = function (code) {
 
 Reply.prototype.status = Reply.prototype.code
 
+Reply.prototype.getSerializationFunction = function (schemaOrStatus) {
+  let serialize
+
+  if (typeof schemaOrStatus === 'string' || typeof schemaOrStatus === 'number') {
+    serialize = this.context[kSchemaResponse] != null
+      ? this.context[kSchemaResponse][schemaOrStatus]
+      : null
+  } else if (typeof schemaOrStatus === 'object') {
+    serialize = this.context[kRequestSerializeFns].get(schemaOrStatus)
+  }
+
+  return serialize
+}
+
+Reply.prototype.compileSerializationSchema = function (schema, httpStatus) {
+  const { request } = this
+  const { method, url } = request
+  let serialize = null
+
+  // Check if serialize function already compiled
+  if (this.context[kRequestSerializeFns].has(schema)) {
+    serialize = this.context[kRequestSerializeFns].get(schema)
+  } else {
+    const { serializerCompiler } = this.context.config
+
+    // If the SchemaController.serializer has not been set for the request
+    // We prepare it here and use it right away
+    if (serializerCompiler == null) {
+      this.server[kSchemaController].setupSerializer(
+        this.server[kOptions]
+      )
+      this.context.config.serializerCompiler = this.server[
```

Comment:
I was going in this direction to kinda tight the SerializerCompiler to the route config, but now that you pointed it out you're right. This can lead to unexpected behaviors for plugins or whoever is relying on the route config, I'll roll it back.